aten/src/ATen/test/tensor_interop_test.cpp

```diff
@@ -0,0 +1,140 @@
+#include "gtest/gtest.h"
+
+#include "ATen/ATen.h"
+#include <caffe2/core/init.h>
+#include <caffe2/core/operator.h>
+
+TEST(TestTensorInterop, Caffe2ToPytorchSimple) {
+  caffe2::Tensor c2_tensor(caffe2::CPU);
+  c2_tensor.Resize(4, 4);
+  auto data = c2_tensor.mutable_data<int64_t>();
+  for (int64_t i = 0; i < 16; i++) {
+    data[i] = i;
+  }
+  at::Tensor at_tensor(c2_tensor.getIntrusivePtr());
+  at::TensorImpl* impl = at_tensor.unsafeGetTensorImpl();
+
+  auto it = impl->data<int64_t>();
+  for (int64_t i = 0; i < 16; i++) {
+    ASSERT_EQ(it[i], i);
+  }
+}
+
+TEST(TestTensorInterop, Caffe2ToPytorchOp) {
+  caffe2::Tensor c2_tensor(caffe2::CPU);
+  c2_tensor.Resize(3, 3);
+  auto data = c2_tensor.mutable_data<int64_t>();
+  for (int64_t i = 0; i < 9; i++) {
+    data[i] = i;
+  }
+  at::Tensor at_tensor(c2_tensor.getIntrusivePtr());
+
+  ASSERT_EQ(at::sum(at_tensor).item<int64_t>(), 36);
+}
+
+TEST(TestTensorInterop, Caffe2ToPytorchUnsupportedDevice) {
+  caffe2::Tensor c2_tensor(caffe2::IDEEP);
+  at::Tensor at_tensor(c2_tensor.getIntrusivePtr());
+  ASSERT_ANY_THROW(at::sum(at_tensor));
+}
+
+TEST(TestTensorInterop, PytorchToCaffe2Op) {
+  caffe2::Workspace workspace;
+  caffe2::NetDef net;
+
+  {
+    auto* blob_a = BlobGetMutableTensor(workspace.CreateBlob("a"), caffe2::CPU);
```

Comment:
We want to get rid of this API as well, please use this instead: https://phabricator.internmc.facebook.com/diffusion/FBS/browse/master/fbcode/caffe2/caffe2/core/blob.h;2acc668cd5c5c19d680290eaf819246f67c023ce$86, we'll rename XBlobGetMutableTensor to BlobGetMutableTensor at some point.